website/static/js/folderPicker.js

```diff
@@ -75,8 +75,10 @@ function _treebeardSelectView(item) {
         return templateUnchecked;
     }
 
-    if (item.data.path === tb.options.folderPath || (tb.options.folderArray && tb.options.folderArray[tb.options.folderArray.length - 1] === item.data.name)) {
```

Comment:
Given this check, every select in the figshare Treebeard grid was getting templateChecked. This means none of the selects have the onclick or onchange events bound to templateUnchecked, and the onPickFolder event is never getting triggered.